### ISSUE TITLE
Cognitive Services: support for `Luis.Authoring`

### ISSUE THREAD

**Issue by tommyJimmy87** [terraform-providers/terraform-provider-azurerm#4319]
Would be cool to have the possibility to add the new Luis Authoring Cognitive Service and also the Web App Bot as possible resources.

**Comment by tombuildsstuff** [comment]
hi @tommyJimmy87 

Thanks for opening this issue :)

Taking a look into this I believe that LUIS support is available via [the `kind` field within the `azurerm_cognitive_account ` resource](https://www.terraform.io/docs/providers/azurerm/r/cognitive_account.html) - would you be able to take a look and see if that works for you? In addition support for Bots has been merged and [will form part of the v1.34.0 release of the Azure Provider](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/CHANGELOG.md). Since this functionality appears to be supported I'm going to close this issue for the moment - however please let us know if that doesn't work for you and we'll take another look.

Thanks!

**Comment by tommyJimmy87** [comment]
@tombuildsstuff  
Hi thanks for the update about the Bot Service. 

Regarding the LUIS Authoring is a different LUIS Cognitive Service where you can choose the Authoring API which will allow to share applications across different accounts in your organization. From the LUIS cognitive service kind in terraform I cannot see the option to choose the Authoring API.

https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-concept-keys

<img width="944" alt="Screen Shot 2019-09-13 at 16 56 38" src="https://user-images.githubusercontent.com/21197220/64872502-9c036100-d647-11e9-88bb-54287a0cc43b.png">

**Comment by tombuildsstuff** [comment]
@tommyJimmy87 so that we could understand that better, would you be able to show the generated ARM Template for that (you should be able to see it by clicking "Automated/Template Deployment" at the bottom of the create screen)? Thanks!

**Comment by tommyJimmy87** [comment]
@tombuildsstuff Sure.

```{
    "$schema": "http://schema.management.azure.com/schemas/2014-04-01-preview/deploymentTemplate.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {
        "name": {
            "type": "string"
        },
        "authoringLocation": {
            "type": "string"
        },
        "authoringSku": {
            "type": "string"
        }
    },
    "resources": [
        {
            "apiVersion": "2016-02-01-preview",
            "name": "[parameters('name')]",
            "location": "[parameters('authoringLocation')]",
            "type": "Microsoft.CognitiveServices/accounts",
            "kind": "LUIS.Authoring",
            "sku": {
                "name": "[parameters('authoringSku')]"
            },
            "properties": {
                "customSubDomainName": "[parameters('name')]"
            }
        }
    ]
}```

**Comment by tommyJimmy87** [comment]
@tombuildsstuff 
I guess is a bit confusing from the Azure guys because if you see from the marketplace they have the same service replicated but with the first one you can create the LUIS app and with the second you can specify which API to use (Authoring, runtime.... )

<img width="1364" alt="Screen Shot 2019-09-13 at 18 21 02" src="https://user-images.githubusercontent.com/21197220/64878389-b5aaa580-d653-11e9-96eb-05c3eba08ba4.png">

**Comment by tombuildsstuff** [comment]
hey @tommyJimmy87 

Thanks for confirming that; it looks like this new option just needs to be added to the `azurerm_cognitive_services` resources list then - as such I've tagged this as such :)

Thanks!

**Comment by enricsoler** [comment]
Hi @tombuildsstuff If it's ok with you, I would like to try to do this, is it ok?

**Comment by katbyte** [comment]
fixed by #4888
